modules/sql-engine/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -527,16 +527,176 @@ SqlCreate SqlCreateZone(Span s, boolean replace) :
         final boolean ifNotExists;
         final SqlIdentifier id;
         SqlNodeList optionList = null;
+        SqlNodeList storageProfiles = null;
 }
 {
     <ZONE> { s.add(this); }
         ifNotExists = IfNotExistsOpt()
         id = CompoundIdentifier()
-    [
-        <WITH> { s.add(this); } optionList = CreateZoneOptionList()
-    ]
+        (
+            <WITH> { s.add(this); } optionList = CreateZoneOptionList()
+            {
+                return new IgniteSqlCreateZone(s.end(this), ifNotExists, id, optionList);
+            }
+            |
+            (
+                [ optionList = ZoneOptionsList() ]
+
+                <STORAGE> <PROFILES> {
+                    storageProfiles = StorageProfiles();
+                }
+            )
+            {
+                SqlIdentifier key = new SqlIdentifier(ZoneOptionEnum.STORAGE_PROFILES.name(), getPos());
+                IgniteSqlZoneOption profilesOption = new IgniteSqlZoneOption(key, storageProfiles, getPos());
+                if (optionList != null) {
+                    optionList.add(profilesOption);
```

Comment:
storage profile is mandatory attribute. Let's have dedicated field in `IgniteSqlCreateZone`